src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/ApartmentValueFunction.java

```diff
@@ -647,84 +589,38 @@ public ApartmentValueFunction adaptWeight(Criterion moreImportant, Criterion les
   }
 
   /**
-   * Gives the subjective value weight of a criterion awt
+   * Gives the subjective value weight of a criterion <code>criterion</code>
    *
-   * @param awt the criterion we want to know the value
+   * @param criterion the criterion we want to know the value
    * @return the subjective value weight
    */
-  double getSubjectiveValueWeight(Criterion awt) {
-    switch (awt) {
-      case TELE:
-        return teleSubjectiveValueWeight;
-      case TERRACE:
-        return terraceSubjectiveValueWeight;
-      case WIFI:
-        return wifiSubjectiveValueWeight;
-      case FLOOR_AREA:
-        return floorAreaSubjectiveValueWeight;
-      case FLOOR_AREA_TERRACE:
-        return floorAreaTerraceSubjectiveValueWeight;
-      case NB_BATHROOMS:
-        return nbBathroomsSubjectiveValueWeight;
-      case NB_BEDROOMS:
-        return nbBedroomsSubjectiveValueWeight;
-      case NB_SLEEPING:
-        return nbSleepingSubjectiveValueWeight;
-      case NB_MIN_NIGHT:
-        return nbMinNightSubjectiveValueWeight;
-      case PRICE_PER_NIGHT:
-        return pricePerNightSubjectiveValueWeight;
-      default:
-        throw new IllegalArgumentException();
-    }
+  double getWeightSubjectiveValue(Criterion criterion) {
+    checkArgument(this.weight.containsKey(criterion));
+    return this.weight.get(criterion);
+  }
+
+  /**
+   * Update the subjective value weight of a criterion <code>criterion</code>
+   *
+   * @param criterion the criterion we want to set the value
+   */
+  public void setWeightSubjectiveValue(final Criterion criterion, final double value) {
```

Comment:
Sans final, par cohérence. OPT.

Reply:
Fix in f338d5ca96f1cc2ad77929e19a335c8c2f905771